### PR TITLE
Profile for Ledger Live desktop app

### PR DESCRIPTION
**Pull request by haplo**
This is a profile for the [Ledger Live wallet](https://www.ledger.com/ledger-live).

I based it off the profile offered in #3595, but it needed several tweaks for it to work.

I tested the binary installation in /opt (I use the [ledger-live-bin package in AUR](https://aur.archlinux.org/packages/ledger-live-bin)). Ledger Live is also offered as an AppImage, but I didn't test that case.

I own a Ledger USB wallet and this profile is working for me.

### CODE REVIEW

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+novideo
+protocol unix,inet,inet6,netlink
+seccomp !chroot
+tracelog
+
+disable-mnt
+private-cache
+# enabling private-dev blocks USB hardware wallets
+# private-dev
+private-etc @network,@tls-ca,@x11
```
Comment (haplo) [inline etc/profile-a-l/ledger-live-desktop.profile:39]: I' not running firejail from master so my actual profile uses:

```
private-etc alternatives,ca-certificates,crypto-policies,host.conf,nsswitch.conf,pki,resolv.conf,rpc,selinux,ssl
```

Networking breaks without these options.

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+nosound
+novideo
+protocol unix,inet,inet6,netlink
+seccomp !chroot
+tracelog
+
+disable-mnt
+private-cache
+# enabling private-dev blocks USB hardware wallets
+# private-dev
```
Comment (haplo) [inline etc/profile-a-l/ledger-live-desktop.profile:38]: I understand not having `private-dev` is a big security hole, but it's the only way I found to get the USB wallet to work. Is there a way to be more fine-grained?

File: etc/profile-a-l/ledger-live-desktop.profile (hunk 1 above)
Comment (glitsj16) [reply etc/profile-a-l/ledger-live-desktop.profile:39]: Good you referenced your private-etc in 0.9.72-format. It translates as private-etc @network,@tls-ca,@x11,host.conf,rpc so you'll need to append `,host.conf,rpc` here for master.

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+include globals.local
+
+noblacklist ${HOME}/.config/Ledger Live
+
+mkdir ${HOME}/.config/Ledger Live
+whitelist ${HOME}/.config/Ledger Live
+whitelist ${DOWNLOADS}
+whitelist /opt/ledger-live
+include whitelist-common.inc
+include whitelist-usr-share-common.inc
```
Comment (glitsj16) [inline etc/profile-a-l/ledger-live-desktop.profile:15]: For alphabetical ordering, move wusc below wrc please.
Can we also add `whitelist run-common.inc`?

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+include whitelist-var-common.inc
+
+apparmor
+caps.drop all
+ipc-namespace
+machine-id
+netfilter
+nodvd
+nogroups
+nonewprivs
```
Comment (glitsj16) [inline etc/profile-a-l/ledger-live-desktop.profile:26]: I'm assuming the profile won't break when adding `noprinters`, but I can't test that.

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+noroot
+notv
```
Comment (glitsj16) [inline etc/profile-a-l/ledger-live-desktop.profile:28]: Alphabetical ordering nitpick. Move this below nosound please.

File: etc/profile-a-l/ledger-live-desktop.profile (hunk 2 above)
Comment (glitsj16) [reply etc/profile-a-l/ledger-live-desktop.profile:38]: The breakage probably is due to your use of the AppImage. We have a conditional for that use case, so the same profile can support both:
`?HAS_APPIMAGE: ignore private-dev`
private-dev

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+private-cache
+# enabling private-dev blocks USB hardware wallets
+# private-dev
+private-etc @network,@tls-ca,@x11
+private-lib
+private-tmp
+
+# app attempts to connect to dbus but seems to work fine when blocked
+dbus-user none
+dbus-system none
```
Comment (glitsj16) [inline etc/profile-a-l/ledger-live-desktop.profile:45]: Can we add `restrict-namespaces`?

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# Firejail profile for Ledger Live
+# This file is overwritten after every install/update
+# Persistent local customizations
+include ledger-live-desktop.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.config/Ledger Live
+
```
Comment (glitsj16) [inline etc/profile-a-l/ledger-live-desktop.profile:9]: You forgot all the `disable-*.inc` includes... Follow the template and double-check if the app still has its regular functionalities please.

```
[...]
# remove disable-write-mnt.inc if you set disable-mnt
include disable-common.inc
include disable-devel.inc
include disable-exec.inc
include disable-interpreters.inc
include disable-proc.inc
include disable-programs.inc
include disable-shell.inc
include disable-write-mnt.inc
include disable-X11.inc
include disable-xdg.inc
[...]
```

**Review by glitsj16 — CHANGES_REQUESTED** [review]
The profile is missing some vitals. See individual comments.

File: etc/profile-a-l/ledger-live-desktop.profile (hunk 2 above)
Comment (haplo) [reply etc/profile-a-l/ledger-live-desktop.profile:54]: I'm using the _/opt/ledger-live_ installation, not the appimage. What I meant is that I haven't tested the appimage release.

Enabling `private-dev` breaks the hardware wallet integration, even after removing `nou2f`.

File: etc/profile-a-l/ledger-live-desktop.profile (hunk 7 above)
Comment (haplo) [reply etc/profile-a-l/ledger-live-desktop.profile:10]: Sorry for this oversight, initially I was including `electron.profile`, but eventually removed it and forgot to add the disables.

I have added them now, will comment on the issues I had in separate threads.

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+# Added by disable-exec.inc, breaks hardware wallet manager
+ignore noexec /tmp
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-proc.inc
```
Comment (haplo) [inline etc/profile-a-l/ledger-live-desktop.profile:18]: This causes these error logs, but the app seems to work fine:

```
[15:0222/105618.284611:ERROR:file_path_watcher_inotify.cc(822)] Failed to read /proc/sys/fs/inotify/max_user_watches
[1:0222/105618.286875:ERROR:file_path_watcher_inotify.cc(822)] Failed to read /proc/sys/fs/inotify/max_user_watches
<----------CUT----------->
[12:0222/105618.665653:ERROR:file_path_watcher_inotify.cc(822)] Failed to read /proc/sys/fs/inotify/max_user_watches
```

File: etc/profile-a-l/ledger-live-desktop.profile (hunk 7 above)
Comment (haplo) [reply etc/profile-a-l/ledger-live-desktop.profile:10]: `disable-X11.inc` causes the app not to start with this error log:

```
[12:0222/101944.253536:ERROR:ozone_platform_x11.cc(240)] Missing X server or $DISPLAY
```

File: etc/profile-a-l/ledger-live-desktop.profile (hunk 4 above)
Comment (haplo) [reply etc/profile-a-l/ledger-live-desktop.profile:40]: I added `noprinters` and it seems to work fine.

**Comment by haplo** [comment]
Thank you very much for the review @glitsj16. I believe I have addressed all your comments. The updated profile is working for me.

**Review by glitsj16 — APPROVED** [review]
Thanks for the changes. LGTM.

File: etc/profile-a-l/ledger-live-desktop.profile [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+novideo
+protocol unix,inet,inet6,netlink
+seccomp !chroot
+tracelog
+
+disable-mnt
+private-cache
+# enabling private-dev blocks USB hardware wallets, if you don't need access to
+# USB devices you can add private-dev to your ledger-live-desktop.local
+# private-dev
```
[the remark this answers is not in the document]
Comment (haplo) [reply etc/profile-a-l/ledger-live-desktop.profile]: > For clarity, use `#` for code and `# ` for normal comments (see #5987).

Good point, I fixed it and will keep it in mind going forward.

**Comment by netblue30** [comment]
merged, thanks!

**Pull request merged**